var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -59,13 +65,31 @@ def headers(self):
         include_path = join_path(self.component_path, 'include')
         return find_headers('*', include_path)
 
+    # provide cluster libraries if +cluster variant is used or
+    # the scalapack virtual package was requested
+    def cluster(self):
+        return '+cluster' in self.spec or 'scalapack' in self.spec
+
     @property
     def libs(self):
-        mkl_libs = [self.xlp64_lib('libmkl_intel'), 'libmkl_sequential', 'libmkl_core']
+        shared = '+shared' in self.spec
+        mkl_libs = []
+        if self.cluster():
+            mkl_libs += [self.xlp64_lib('libmkl_scalapack'),
+                         'libmkl_cdft_core']
+        mkl_libs += [self.xlp64_lib('libmkl_intel'),
+                     'libmkl_sequential',
+                     'libmkl_core']
+        if self.cluster():
+            mkl_libs += [self.xlp64_lib('libmkl_blacs_intelmpi')]
         libs = find_libraries(mkl_libs,
-                              join_path(self.component_path, 'lib', 'intel64'))
-        libs += find_system_libraries(['libpthread', 'libm', 'libdl'])
-        return libs
+                              join_path(self.component_path, 'lib', 'intel64'),
+                              shared=shared)
+        system_libs = find_system_libraries(['libpthread', 'libm', 'libdl'])
+        if shared:
+            return libs + system_libs
+        else:
+            return IntelOneApiStaticLibraryList(libs, system_libs)
```

Comment:
(question: no change needed) if I understand correctly, this is a stopgap pending the introduction of "use variants" (https://github.com/spack/seps/pull/2)

Reply:
Yes, this should be handled by a use variant